internal/venice-common/src/main/java/com/linkedin/venice/writer/VeniceWriter.java

```diff
@@ -113,6 +113,20 @@ public class VeniceWriter<K, V, U> extends AbstractVeniceWriter<K, V, U> {
   public static final String MAX_ELAPSED_TIME_FOR_SEGMENT_IN_MS =
       VENICE_WRITER_CONFIG_PREFIX + "max.elapsed.time.for.segment.in.ms";
 
+  /**
+   * Maximum Venice record size. Default: {@value DEFAULT_MAX_RECORD_SIZE_BYTES}
+   *
+   * Large records can cause performance issues, so this setting is used to detect and prevent them. Not to be
+   * confused with Kafka record size (which is the ~1MB limit {@link MAX_SIZE_FOR_USER_PAYLOAD_PER_MESSAGE_IN_BYTES}
+   * is designed to comply with). Venice records refer to a Venice key-value pair, which can be spread across 1+ Kafka
+   * records / events. Basically: Record without Chunking < 1 MB < Record with Chunking < Max Record Size
+   *
+   * 1. If a batch push data contains records larger than this setting, the push job will fail.
+   * 2. If a nearline job contains records larger than this setting, consumption will be paused and manual intervention
+   * will be necessary.
+   */
+  // public static final String MAX_RECORD_SIZE_BYTES = VENICE_WRITER_CONFIG_PREFIX + "max.record.size.bytes";
```

Comment:
These lines have to be commented out, otherwise the code coverage tool is going to complain. They will be used in the upcoming changes.